src/AdvApi32.Desktop/AdvApi32.cs

```diff
@@ -888,14 +933,169 @@ public static partial class AdvApi32
             int cbBufSize,
             out int pcbBytesNeeded);
 
         /// <summary>
+        /// Registers a function to handle service control requests.
+        /// </summary>
+        /// <param name="lpServiceName">
+        /// The name of the service run by the calling thread. This is the service name that the service control program specified in the CreateService function when creating the service.
+        /// If the service type is SERVICE_WIN32_OWN_PROCESS, the function does not verify that the specified name is valid, because there is only one registered service in the process.
+        /// </param>
+        /// <param name="lpHandlerProc">A reference to the handler function to be registered.</param>
+        /// <returns>If the function succeeds, the return value is a service status handle If the function fails, the return value is zero. </returns>
+        /// <remarks>
+        /// This function has been superseded by the <see cref="RegisterServiceCtrlHandlerEx"/> function.
+        /// A service can use either function, but the new function supports user-defined context data, and the new handler function supports additional extended control codes.
+        /// </remarks>
+        [DllImport(nameof(AdvApi32), CharSet = CharSet.Unicode)]
+        [Obsolete("Use LPHANDLER_FUNCTION_EX with RegisterServiceCtrlHandlerEx instead")]
+#pragma warning disable CS0618 // Type or member is obsolete
+        public static extern IntPtr RegisterServiceCtrlHandler(string lpServiceName, LPHANDLER_FUNCTION lpHandlerProc);
+#pragma warning restore CS0618 // Type or member is obsolete
+
+        /// <summary>
+        /// Registers a function to handle extended service control requests.
+        /// </summary>
+        /// <param name="lpServiceName">
+        /// The name of the service run by the calling thread. This is the service name that the service control program specified in the CreateService function when creating the service.
+        /// </param>
+        /// <param name="lpHandlerProc">A reference to the handler function to be registered.</param>
+        /// <param name="lpContext">Any user-defined data. This parameter, which is passed to the handler function, can help identify the service when multiple services share a process.</param>
+        /// <returns>If the function succeeds, the return value is a service status handle If the function fails, the return value is zero. </returns>
+        /// <remarks>
+        /// This function has been superseded by the <see cref="RegisterServiceCtrlHandlerEx"/> function.
+        /// A service can use either function, but the new function supports user-defined context data, and the new handler function supports additional extended control codes.
+        /// </remarks>
+        [DllImport(nameof(AdvApi32), CharSet = CharSet.Unicode)]
+        public static extern System.IntPtr RegisterServiceCtrlHandlerEx(string lpServiceName, LPHANDLER_FUNCTION_EX lpHandlerProc, IntPtr lpContext);
```

Comment:
I agree with your reasoning. Handles should either be SafeHandles (when the caller should dispose them) or IntPtrs.